core/src/avm1/globals/bevel_filter.rs

```diff
@@ -0,0 +1,676 @@
+
+    let clamped_angle = if angle.is_sign_negative() {
+        -(angle.abs() % 360.0)
+    } else {
+        angle % 360.0
+    };
+
+    let highlight_color = args
+        .get(2)
+        .unwrap_or(&0xFFFFFF.into())
+        .coerce_to_i32(activation)?;
+
+    let highlight_color_clamped = if highlight_color.is_negative() {
+        0x1000000 - (highlight_color.abs() % 0x1000000)
+    } else {
+        highlight_color % (0x1000000)
+    };
+
+    let highlight_alpha = args
+        .get(3)
+        .unwrap_or(&1.into())
+        .coerce_to_f64(activation)
+        .map(|x| x.max(0.0).min(1.0))?;
+
+    let shadow_color = args
+        .get(4)
+        .unwrap_or(&0x000000.into())
+        .coerce_to_i32(activation)?;
+
+    let shadow_color_clamped = if shadow_color.is_negative() {
+        0x1000000 - (shadow_color.abs() % 0x1000000)
+    } else {
+        shadow_color % (0x1000000)
+    };
+
+    let shadow_alpha = args
+        .get(5)
+        .unwrap_or(&1.into())
+        .coerce_to_f64(activation)
+        .map(|x| x.max(0.0).min(1.0))?;
+
+    let blur_x = args
+        .get(6)
+        .unwrap_or(&4.into())
+        .coerce_to_f64(activation)
+        .map(|x| x.max(0.0).min(255.0))?;
+
+    let blur_y = args
+        .get(7)
+        .unwrap_or(&4.into())
+        .coerce_to_f64(activation)
+        .map(|x| x.max(0.0).min(255.0))?;
+
+    let strength = args
+        .get(8)
+        .unwrap_or(&1.into())
+        .coerce_to_f64(activation)
+        .map(|x| x.max(0.0).min(255.0))?;
+
+    let quality = args
+        .get(9)
+        .unwrap_or(&1.into())
+        .coerce_to_i32(activation)
+        .map(|x| x.max(0).min(15))?;
+
+    let type_: BevelFilterType = args
+        .get(10)
+        .unwrap_or(&"inner".into())
+        .coerce_to_string(activation)
+        .map(|v| v.as_str().into())?;
+
+    let knockout = args
+        .get(11)
+        .unwrap_or(&false.into())
+        .as_bool(activation.current_swf_version());
+
+    let bevel_filter = this.as_bevel_filter_object().unwrap();
+
+    bevel_filter.set_distance(activation.context.gc_context, distance);
+    bevel_filter.set_angle(activation.context.gc_context, clamped_angle);
+    bevel_filter.set_highlight_color(activation.context.gc_context, highlight_color_clamped);
+    bevel_filter.set_highlight_alpha(activation.context.gc_context, highlight_alpha);
+    bevel_filter.set_shadow_color(activation.context.gc_context, shadow_color_clamped);
+    bevel_filter.set_shadow_alpha(activation.context.gc_context, shadow_alpha);
+    bevel_filter.set_blur_x(activation.context.gc_context, blur_x);
+    bevel_filter.set_blur_y(activation.context.gc_context, blur_y);
+    bevel_filter.set_strength(activation.context.gc_context, strength);
+    bevel_filter.set_quality(activation.context.gc_context, quality);
+    bevel_filter.set_type(activation.context.gc_context, type_);
+    bevel_filter.set_knockout(activation.context.gc_context, knockout);
```

Comment:
If these setters are going to clamp the values, do we have to do it above too? Seems like we're doing it twice